docs/tutorials/telescope_prog_intro.md

```diff
@@ -0,0 +1,247 @@
+        logging.warn("Cleanup task executing")
+```
+
+This telescope adds `task1, task2, cleanup` tasks that just print some statements. The function signature is always the same for Telescope tasks.   If you need to pass additional arguments, you should do it through keyword arguments, and process `**kwargs` within the task function.
+
+If you start the observatory platform with these changes, you will see a new DAG in Airflow called **my_dag_id** with the DAG structure `task1 -> task2 -> cleanup`.  When you run the DAG, you should see the logging messages in the log of each task.
+
+#### BigQuery schemas
+
+BigQuery database schema json files are put in `observatory-dags/dags/database/schema`.  They follow the scheme: `<table_name>_YYYY-MM-DD.json`.  If you wish to provide an additional custom version as well as the date, then the files should follow the scheme: `<table_name>_customversion_YYYY-MM-DD.json`.
+
+The BigQuery table loading utility functions in the Academic Observatory platform API will try to find the correct schema to use for loading table data, based on release date information.
+
+#### Specialised telescopes
+
+Currently there are two types of specialised telescope patterns implemented by the Academic Observatory.  These are:
+```eval_rst
+#. the `StreamTelescope`, where the datasets have an initial full snapshot, followed by differential releases, and
+#. the `SnapshotTelescope`, where each release is a full snapshot of the data.
+```
+
+The stream and snapshot telescopes derive the `Telescope` class.
+
+These telescopes provide extra methods for loading data into BigQuery.  Currently only Google Cloud Storage and BigQuery are supported.
+
+If you wish to design your own telescope templates, you can choose to derive the `Telescope` class if it suits your needs, or implement the `AbstractTelescope` interface yourself.
+
+### Generating a telescope template
+
+You can use the observatory cli tool to generate a telescope template for a new `Telescope`, `StreamTelescope` or `SnapshotTelescope` telescope and release class. Use the command:
+```
+observatory generate telescope <type> <name>
+```
+where the type can be `Telescope`, `StreamTelescope`, `SnapshotTelescope` and name is the class name of your new telescope.  It will generate a new dag `.py` file in `observatory-dags/observatory/dags/dags` with the lower case class name.  Similarly a telescope template will be created in `observatory-dags/observatory/dags/telescopes` with the same lower case class name.
+
+For example:
+```
+observatory generate telescope SnapshotTelescope MyNewTelescope
+```
+creates the files `observatory-dags/observatory/dags/dags/mynewtelescope.py` and `observatory-dags/observatory/dags/telescopes/mynewtelescope.py`
+
+### Testing
+
+You can test telescopes by running them in Airflow.  The Academic Observatory has additional infrastructure to facilitate unit testing of telescopes without needing to run the telescopes in Airflow.
```

Comment:
I think either remove this section, or provide a link to the full Testing section.